### PR TITLE
Print classify percentages up to two significant digits

### PR DESCRIPTION
**Pull request by michalpalka**
Currently, the percentage values of occurrences of labels
in test cases are printed using integral percentages, which
is not precise enough if the labels occur very rarely. This
commit prints them up to two siginificant digits.

### CODE REVIEW

**Comment by michalpalka** [comment]
Actually, let me refactor this-I'll submit a new version soon.

File: Test/QuickCheck/Test.hs [hunk 1]
```diff
@@ -361,20 +363,41 @@ success st =
               , not (null s')
               ]
 
-  covers = [ ("only " ++ show (labelPercentage l st) ++ "% " ++ l ++ ", not " ++ show reqP ++ "%")
+  covers = [ ("only " ++ show (labelPrecise (labelCount l st) (numSuccessTests st)) ++ "% " ++ l ++ ", not " ++ show reqP ++ "%")
            | (l, reqP) <- Map.toList (S.labels st)
            , labelPercentage l st < reqP
            ]
 
-  showP p = (if p < 10 then " " else "") ++ show p ++ "% "
```
Comment (michalpalka) [inline Test/QuickCheck/Test.hs]: I am not sure what was the purpose of this, but I left it as close to the original functionality as I could.

**Comment by nick8325** [comment]
It seems to no longer print the labels in decreasing order of frequency.

I think it should be possible to do this simpler - I'll have a go at it.

**Comment by sol** [comment]
How will this relate to the `labels` field of the `Result` type?  Does it represent the same information?  Will that still be the case after this PR?

I'm asking because it's type is `[(String, Int)]`, so I would have assumed we need to change that too, if we want to increase the precision.

**Comment by nick8325** [comment]
Yes, changing the result type to match seems like a good idea...

**Pull request closed without merging**